packages/react-native-reanimated/plugin/src/workletFactory.ts

```diff
@@ -325,17 +325,44 @@ function hash(str: string): number {
   return (hash1 >>> 0) * 4096 + (hash2 >>> 0);
 }
 
-function makeWorkletName(fun: NodePath<WorkletizableFunction>): string {
+let functionId = 1;
+
+function makeWorkletName(
+  fun: NodePath<WorkletizableFunction>,
+  state: ReanimatedPluginPass
+): string {
+  let source = 'unknown_file';
+
+  if (state.file.opts.filename) {
+    const filepath = state.file.opts.filename;
+    // Take the file name without extension.
+    source = basename(filepath).split('.')[0];
+    const splitFilepath = filepath.split('/');
+    // Get the library name from the path.
+    const nodeModulesIndex = splitFilepath.indexOf('node_modules');
+    if (nodeModulesIndex !== -1) {
+      // Remove all non-alphanumeric characters.
+      const libraryName = splitFilepath[nodeModulesIndex + 1].replace(
+        /\W/g,
+        ''
+      );
+      source = libraryName + '_' + source;
+    }
+  }
+
+  const suffix = '_' + source + functionId++;
   if (isObjectMethod(fun.node) && isIdentifier(fun.node.key)) {
-    return fun.node.key.name;
+    return fun.node.key.name + suffix;
   }
   if (isFunctionDeclaration(fun.node) && isIdentifier(fun.node.id)) {
-    return fun.node.id.name;
+    return fun.node.id.name + suffix;
   }
   if (isFunctionExpression(fun.node) && isIdentifier(fun.node.id)) {
-    return fun.node.id.name;
+    return fun.node.id.name + suffix;
   }
-  return 'anonymous'; // fallback for ArrowFunctionExpression and unnamed FunctionExpression
+
+  // Fallback for ArrowFunctionExpression and unnamed FunctionExpression.
+  return suffix;
```

Comment:
`suffix` starts with an underscore, can we get rid of it here?
```ts
const suffix = `${source}${functionId++}`;

return `${fun.node.key.name}_${suffix}`;
```

Reply:
@tomekzaw Done!